harness/reporters/reporter.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reporters
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/coreos/pkg/multierror"
+)
+
+type Reporters []Reporter
+
+func (reps *Reporters) ReportTest(name, result string, duration time.Duration, b []byte) {
+	for _, r := range *reps {
+		r.ReportTest(name, result, duration, b)
+	}
+}
+
+func (reps *Reporters) Cleanup() error {
+	var err multierror.Error
+
+	for _, r := range *reps {
+		e := r.Cleanup()
+		if e != nil {
+			fmt.Println(e)
+			err = append(err, e)
+		}
+	}
+
+	return err.AsError()
+}
+
+func (reps *Reporters) Output() error {
+	for _, r := range *reps {
+		err := r.Output()
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (reps *Reporters) SetResult(s string) {
+	for _, r := range *reps {
+		r.SetResult(s)
+	}
+}
+
+func (reps *Reporters) OpenFile(pathFunc func(string) string) error {
```

Comment:
I'd much rather this be passed a `outputRoot` or something and then figure out the right name based on that.

I'm also not a fan of `OpenFile` as the name. Something like `SetupOutputDir` might be better?

I'm also dubious of the value of not just opening the file in the `Output` function. The file handle isn't being used between this "OpenFile" and the eventual write of output.

I'm not sure that the earlier error reporting / failure is worth this additional complexity

Reply:
I don't have a particular strong opinion either way, the reason that I have the `OpenFile` seperated from the `Output` function is to be able to provide similar functionality to what happens with the Profile's. Inside of the `harness/suite` they have their file handlers opened directly inside of the `Run` function while the actual output is defer'd (see https://github.com/coreos/mantle/blob/master/harness/suite.go#L208).

If we are going to change these to open the file handler inside of the `Output` method do we want to move the profiles to also opening their files inside of the defer'd function?